test/mount/extensions/explicit_deps_test.clj

```diff
@@ -40,6 +57,15 @@
     (deps/start #'state-3))
   (is (status) {#'state-1 :stopped #'state-2 :started #'state-3 :started}))
 
+(deftest start-with-overridden-dependency
```

Comment:
Substitutes should always be non-global, i.e. not defined using `defstate` but by using `state`. Furthermore, all `defstate`s one intends to use should be loaded on application boot-up. Defining new a `defstate` later is not how mount-lite should be used. Could you rewrite this test to reflect that?